### PR TITLE
C++17 fixes

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

File: python/pythonizations/PyROOT_smartptrtest.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def test01_transparency(self):
         mine = cppyy.gbl.mine
 
         assert type(mine) == MyShareable
-        assert type(mine._get_smart_ptr()) == cppyy.gbl.std.auto_ptr(MyShareable)
+        assert type(mine._get_smart_ptr()) == cppyy.gbl.std.shared_ptr(MyShareable)
```
Comment (pcanal) [inline python/pythonizations/PyROOT_smartptrtest.py]: this seems to need more change to the script ....

Comment (Axel-Naumann) [reply python/pythonizations/PyROOT_smartptrtest.py]: I see the failures, but I don't understand them :-/

Comment (pcanal) [reply python/pythonizations/PyROOT_smartptrtest.py]: Did you change the implementation of _get_smart_ptr?

Comment (Axel-Naumann) [reply python/pythonizations/PyROOT_smartptrtest.py]: No. And I shouldn't have to (if you look at the machinery that makes this work - in the end, `Cppyy::IsSmartPtr()`. More guesses? ;-)

Comment (pcanal) [reply python/pythonizations/PyROOT_smartptrtest.py]: Then I don't understand the semantic of the line ... if nothing else change why would the return of type(mine._get_smart_ptr()) switch from auto_ptr to shared_ptr?

Comment (pcanal) [reply python/pythonizations/PyROOT_smartptrtest.py]: humm .. you also change SmartPtr* ... so okay ...

Comment (pcanal) [reply python/pythonizations/PyROOT_smartptrtest.py]: and I was off by one line on where the error happen, so never mind ....

**Comment by Axel-Naumann** [comment]
Needs https://github.com/root-project/root/pull/3840

**Comment by Axel-Naumann** [comment]
@phsft-bot build

**Pull request merged**